app/src/main/java/com/github/libretube/ui/views/DescriptionLayout.kt

```diff
@@ -40,6 +41,19 @@ class DescriptionLayout(
         }
     }
 
+    fun setSegments(segments: List<Segment>) {
+        segments.filter { it.actionType == Segment.TYPE_FULL }.firstNotNullOfOrNull {
+            when (it.category) {
+                "sponsor" -> context.getString(R.string.category_sponsor)
+                "exclusive_access" -> context.getString(R.string.category_exclusive_access)
+                else -> null
+            }
+        }?.let {
+            binding.playerSponsorBadge.isVisible = true
+            binding.playerSponsorBadge.text = it
+        }
```

Comment:
We should hide the badge when no matching category is found, otherwise it stays visible even though there are no SponsorBlock segments in the next video.

Reply:
I think it should already be hidden, since `android:visibility="gone"` is set?

Reply:
*sigh* Android...

Reply:
android-rs when? :eyes: